clusters/resource_cluster.go

```diff
@@ -383,6 +384,40 @@ func setPinnedStatus(ctx context.Context, d *schema.ResourceData, clusterAPI com
 	return d.Set("is_pinned", pinnedEvent == compute.EventTypePinned)
 }
 
+func removeUnnecessaryFieldsFromForceSendFields(clusterSpec *compute.EditCluster) {
+	if clusterSpec.AwsAttributes != nil {
+		newAwsAttributesForceSendFields := []string{}
+		// These fields should never be 0.
+		unnecessaryFieldNamesForAwsAttributes := []string{
+			"SpotBidPricePercent",
+			"EbsVolumeCount",
+			"EbsVolumeIops",
+			"EbsVolumeSize",
+			"EbsVolumeThroughput",
+		}
+		for _, field := range clusterSpec.AwsAttributes.ForceSendFields {
+			if !slices.Contains(unnecessaryFieldNamesForAwsAttributes, field) {
+				newAwsAttributesForceSendFields = append(newAwsAttributesForceSendFields, field)
+			}
+		}
+		clusterSpec.AwsAttributes.ForceSendFields = newAwsAttributesForceSendFields
+	}
+	if clusterSpec.GcpAttributes != nil {
+		newGcpAttributesForceSendFields := []string{}
+		// These fields should never be 0.
+		unnecessaryFieldNamesForGcpAttributes := []string{
+			"BootDiskSize",
+			"LocalSsdCount",
```

Comment:
LocalSsdCount is valid zero value - we had customer report for omitting it...

Reply:
oops alright let me update